metpy/calc/thermo.py

```diff
@@ -274,6 +314,199 @@ def parcel_profile(pressure, temperature, dewpt):
+            point pressure.
+        temperature : `pint.Quantity`
+            The starting temperature
+        dewpt : `pint.Quantity`
+            The starting dew point
+        parcel_temperature : `pint.Quantity`
+            The temperature of the parcel
+
+        Returns
+        -------
+        `pint.Quantity`
+            Convective available potential energy (CAPE).
+
+        See Also
+        --------
+        virtual_temperature, find_intersections, lfc, el
+        """
+    # Calculate limits of integration
+    # If there is no LFC, no need to proceed. No EL and we use the top reading of the sounding.
+    lfc_pressure = lfc(pressure, temperature, dewpt)[0]
+    el_pressure = el(pressure, temperature, dewpt)[0]
+
+    if lfc_pressure is None:
+        return 0 * units('J/kg')
+    else:
+        lfc_pressure = lfc_pressure.magnitude
+
+    if el_pressure is None:
+        el_pressure = pressure[-1].magnitude
+    else:
+        el_pressure = el_pressure.magnitude
+
+    # If no parcel path is specified, calculated the surface based parcel path
+    if parcel_temperature is None:
+        parcel_temperature = parcel_profile(pressure, temperature[0], dewpt[0])
+
+    # Calculate virtual temperature if requested
+    if use_virtual_temperature:
+        temperature = virtual_temperature(pressure, temperature, dewpt)
+        parcel_temperature = virtual_temperature(pressure, parcel_temperature, dewpt)
+
+    y = (parcel_temperature - temperature).to(units.degK)
+
+    x,y = _find_append_zero_crossings(np.copy(pressure),y)
+
+    # Clip out negative area (temperature parcel < temperature environment)
+    y[y <= 0 * units.degK] = 0 * units.degK
+
+    # Only use data between the LFC and EL for calculation
+
+    p_mask = (x <= lfc_pressure) & (x >= el_pressure)
+    x = x[p_mask]
+    y = y[p_mask]
+
+    return (Rd * (np.trapz(y, np.log(x)) * units.degK)).to(units('J/kg'))
+
+
+@exporter.export
+def cin(pressure, temperature, dewpt, parcel_temperature=None, use_virtual_temperature=False):
+    r"""Calculate convective inhibition (CIN).
+
+        Description
+
+        Parameters
+        ----------
+        pressure : `pint.Quantity`
+            The atmospheric pressure level(s) of interest. The first entry should be the starting
+            point pressure.
+        temperature : `pint.Quantity`
+            The starting temperature
+        dewpt : `pint.Quantity`
+            The starting dew point
+        parcel_temperature : `pint.Quantity`
+            The temperature of the parcel
+
+        Returns
+        -------
+        `pint.Quantity`
+            Convective inhibition (CIN).
+
+        See Also
+        --------
+        virtual_temperature, find_intersections, lfc
+        """
+    # Calculate limits of integration
+    # If there is no LFC, no need to proceed.
+    lfc_pressure = lfc(pressure, temperature, dewpt)[0]
+    if lfc_pressure is None:
+        return 0 * units('J/kg')
```

Comment:
I'm not sure - the "book" definition says the limits of integration on CIN are P0 to pLFC.